infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -258,6 +240,58 @@ private Set<String> readStreamsToReplicateFromTable() {
         return tableNames;
     }
 
+    /**
+     * Helper method for checking whether LR is in upgrading path or not.
+     * Note that the default boolean value for ProtoBuf message is false.
+     *
+     * @return True if LR is in upgrading path, false otherwise.
+     */
+    public boolean isUpgraded() {
+        VersionString versionString = VersionString.newBuilder()
+                .setName(VERSION_PLUGIN_KEY).build();
+        CorfuStoreEntry<VersionString, Version, CommonTypes.Uuid> record;
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            record = txn.getRecord(LOG_REPLICATION_PLUGIN_VERSION_TABLE, versionString);
+            txn.commit();
+        } catch (NoSuchElementException e) {
+            // Normally this will not happen as version table should be initialized during bootstrap
+            log.error("Version table has not been initialized", e);
+            return false;
+        }
+        return record.getPayload().getIsUpgraded();
+    }
+
+    /**
+     * Helper method for flipping the boolean flag in version table which indicates the LR
+     * upgrading path is complete.
+     */
+    public void resetUpgradeFlag(boolean upgraded) {
```

Comment:
didnt we change this method to not take any parameter as it just sets the flag to false?